src/Raven.Server/Documents/ComputeHttpEtags.cs

```diff
@@ -96,6 +96,39 @@ public static unsafe string ComputeEtagForRevisions(List<Document> revisions)
             return str;
         }
 
+        public static unsafe string CombineEtags(IEnumerable<string> etags)
+        {
+
+            var size = Sodium.crypto_generichash_bytes();
+            Debug.Assert((int)size == 32);
+            var cryptoGenerichashStatebytes = (int)Sodium.crypto_generichash_statebytes();
+            byte* state = stackalloc byte[cryptoGenerichashStatebytes];
+            if (Sodium.crypto_generichash_init(state, null, UIntPtr.Zero, size) != 0)
+                ThrowFailToInitHash();
+
+            foreach (var etag in etags)
+            {
```

Comment:
You are passing potential nulls here, that is likely going to kill the server

Reply:
how is it different from what we do with documents:
https://github.com/ravendb/ravendb/blob/feature/sharding/src/Raven.Server/Documents/ComputeHttpEtags.cs#L103

Reply:
There it is known you can't get null.
In you EnumerateEtags, you are expecting nulls